### PR TITLE
Update: unzip

### PR DESCRIPTION
**Pull request by tuncK**
1) Conda package version update v0.2 -> v6.0
2) The current wrapper is unable to handle zip contents if they are not simply named basename.ext and omits them. I tried to find a workaround.

### CODE REVIEW

**Comment by tuncK** [comment]
Ping @bgruening

File: tools/unzip/unzip.xml [hunk 1]
```diff
@@ -1,88 +1,125 @@
...
-          #if $extract_options['extract_all'] == 'False'
-            xfO "$input_file" "$extract_options['pathtofile']" > out.data
-          #else
-            -xf "$input_file" -C ./out/
-          #end if
+            tar
+            #if $extract_options['extract_all'] == 'False'
+              xfO "$input_file" "$extract_options['pathtofile']" > out.data
+            #else
+              -xf "$input_file" -C ./out/
```
Comment (bgruening) [inline tools/unzip/unzip.xml:28]: single quotes

File: tools/unzip/unzip.xml [hunk 2]
```diff
@@ -1,88 +1,125 @@
...
         #if $input_file.is_of_type("zip"):
-          unzip
-          #if $extract_options['extract_all'] == 'False'
-            -p "$input_file" "$extract_options['pathtofile']" > out.data
-          #else
-            "$input_file" -d out/
-          #end if
+            unzip
+            #if $extract_options['extract_all'] == 'False'
+                -p "$input_file" "$extract_options['pathtofile']" > out.data
```
Comment (bgruening) [inline tools/unzip/unzip.xml:18]: singe quotes

File: tools/unzip/unzip.xml [hunk 3]
```diff
@@ -1,88 +1,125 @@
...
-    </conditional>
-  </inputs>
-  <outputs>
-        <data auto_format="true" name="unzipped_single" from_work_dir="out.data">
-          <filter>extract_options['extract_all'] == 'False'</filter>
+        &&
+
+        ## Flatten any potential directory structure, e.g.
+        ## out/sub1/subsub2/fileA.ext -> out/sub1_subsub2_fileA.ext
+        find out -type f | cut -d '/' -f 2- | grep '/' | xargs -I % sh -c "mv out/% out/\\$(echo % | tr / _)"
```
Comment (bgruening) [inline tools/unzip/unzip.xml:35]: do we need gnu-coreutil for find?

Comment (bgruening) [reply tools/unzip/unzip.xml:35]: oh and grep

Comment (tuncK) [reply tools/unzip/unzip.xml:35]: Unless we want to pin the version, gnu-coreutils is not needed. Not much change can happen here I guess?

File: tools/unzip/unzip.xml (hunk 2 above)
Comment (tuncK) [reply tools/unzip/unzip.xml]: Done

File: tools/unzip/unzip.xml (hunk 1 above)
Comment (tuncK) [reply tools/unzip/unzip.xml]: Done

**Comment by bgruening** [comment]
@tuncK can you maybe also update the GHA here? Just take the latest ones from IUC. Thanks.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks!

**Comment by tuncK** [comment]
@bgruening This (unzip) was unfortunately not auto-updated on usegalaxy.eu. It is in collections.yaml (https://github.com/tuncK/usegalaxy-eu-tools/blob/unzip/collections.yaml), but collections.yaml does not seem to be monitored by the update bot (https://github.com/usegalaxy-eu/usegalaxy-eu-tools/blob/6b2fdc47a76e1b3de1d6e5932509c79d0ca73e75/Makefile#L37).

Would shifting unzip to imgteam.yaml be a good fix? https://github.com/usegalaxy-eu/usegalaxy-eu-tools/pull/571

**Comment by bgruening** [comment]
You could just add this specific version hash to the collection.yaml.lock file. That would also make it installable.

**Pull request merged**